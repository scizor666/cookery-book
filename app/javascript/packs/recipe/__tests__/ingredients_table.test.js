import React from 'react';
import {ingredients} from './test_data';
import renderer from 'react-test-renderer';
jest.mock('../ingredient_row', () => 'IngredientRow');
import IngredientsTable from '../ingredients_table';

test('display ingredients table', () => {
    const wrapper = renderer.create(
        <IngredientsTable ingredients={ingredients} caloricity={123.45}/>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
});