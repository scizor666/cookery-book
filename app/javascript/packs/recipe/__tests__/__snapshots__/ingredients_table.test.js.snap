// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display ingredients table 1`] = `
<div
  className="col-sm-12 col-md-6"
>
  <h2>
    Ingredients
  </h2>
  <table
    className="table table-striped lead ingredients-table"
  >
    <thead>
      <tr>
        <th
          className="col"
        >
          Name
        </th>
        <th
          className="col"
        >
          Weight,g
        </th>
        <th
          className="col"
        >
          kcal/100g
        </th>
      </tr>
    </thead>
    <tbody>
      <IngredientRow
        ingredient={
          Object {
            "id": 1,
            "product": Object {
              "caloricity": 107.7,
              "id": 2,
              "name": "chicken",
            },
            "weight": 300,
          }
        }
      />
      <IngredientRow
        ingredient={
          Object {
            "id": 2,
            "product": Object {
              "caloricity": 44,
              "id": 1,
              "name": "carrot",
            },
            "weight": 155,
          }
        }
      />
      <tr>
        <td>
          <h4>
            Total
          </h4>
        </td>
        <td>
          455
        </td>
        <td>
          <h4>
            123.45
          </h4>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
